concent_api/core/views.py

```diff
@@ -45,53 +47,86 @@ def send(_request, client_message):
 @api_view
 @require_POST
 def receive(request, _message):
-    current_time = int(datetime.datetime.now().timestamp())
-    last_undelivered_message_status = ReceiveStatus.objects.filter(delivered = False).order_by('timestamp').last()
+    current_time      = int(datetime.datetime.now().timestamp())
+    client_public_key = decode_client_public_key(request)
+    last_undelivered_message_status = ReceiveStatus.objects.filter_public_key(
+        request.META['HTTP_CONCENT_CLIENT_PUBLIC_KEY']
+    ).filter(delivered = False).order_by('timestamp').last()
     if last_undelivered_message_status is None:
-        last_delivered_message_status = ReceiveStatus.objects.all().order_by('timestamp').last()
+        last_delivered_message_status = ReceiveStatus.objects.filter_public_key(
+            request.META['HTTP_CONCENT_CLIENT_PUBLIC_KEY']
+        ).order_by('timestamp').last()
         if last_delivered_message_status is None:
             return None
-        if last_delivered_message_status.message.type == message.ForceReportComputedTask.TYPE:
-            return handle_receive_delivered_force_report_computed_task(last_delivered_message_status)
-        if last_delivered_message_status.message.type == message.concents.ForceGetTaskResultUpload.TYPE:
+
+        if (
+            last_delivered_message_status.message.type                           == message.ForceReportComputedTask.TYPE and
+            last_delivered_message_status.message.auth.provider_public_key_bytes == client_public_key
+        ):
+            return handle_receive_delivered_force_report_computed_task(request, last_delivered_message_status)
+
+        if (
+            last_delivered_message_status.message.type                            == message.concents.ForceGetTaskResultUpload.TYPE and
+            last_delivered_message_status.message.auth.requestor_public_key_bytes == client_public_key
+        ):
             decoded_message_data = deserialize_message(last_delivered_message_status.message.data.tobytes())
             file_uploaded        = get_file_status(decoded_message_data.file_transfer_token)
             if file_uploaded:
-                return handle_receive_force_get_task_result_upload_for_requestor(decoded_message_data)
+                return handle_receive_force_get_task_result_upload_for_requestor(
+                    request, decoded_message_data, last_delivered_message_status
```

Comment:
Please fix your style here. One parameter per line.

Reply:
Why ? There is nothing about it in our coding style guidelines.

Reply:
Because that's more readable. It's easy to miss that this is more than one parameter. I actually thought it's a single on until I later went back to this line to check something and looked at it closer.

There are examples in our coding style that show each parameter on a separate line. There isn't one showing this case specifically but I honestly did not think that someone would do that :) I'll add it as an example.

Reply:
:+1: I thought that if it is not exceeding a line it is ok, as they are other examples of this in our code, but off course I will change it.

Reply:
If there's something like this somewhere in our code I must have missed it.